module/zfs/dmu_object.c

```diff
@@ -61,6 +61,7 @@ dmu_object_alloc_dnsize(objset_t *os, dmu_object_type_t ot, int blocksize,
 	boolean_t restarted = B_FALSE;
 	uint64_t *cpuobj = NULL;
 	int dnodes_per_chunk = 1 << dmu_object_alloc_chunk_shift;
+	int error;
 
 	kpreempt_disable();
 	cpuobj = &os->os_obj_next_percpu[CPU_SEQID %
```

Comment:
I'm puzzled by this 'per-cpu' usage pattern here. This pointer is dereferenced later on with preemption enabled, letting other threads to use the same value if scheduled on the same CPU. Moreover, the 'per-cpu' value is updated (atomically?) later, which [can race with other threads](http://build.zfsonlinux.org/builders/CentOS%207%20x86_64%20Mainline%20%28TEST%29/builds/3601/steps/shell_9/logs/console) (see #6470). Are these caveats handled somewhere/somehow?